packages/gatsby-plugin-google-tagmanager/src/gatsby-ssr.js

```diff
@@ -1,46 +1,70 @@
 import React from "react"
 import { oneLine, stripIndent } from "common-tags"
 
+const generateGTM = ({ id, environmentParamStr }) => stripIndent`
+  (function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
+  new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
+  j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
+  'https://www.googletagmanager.com/gtm.js?id='+i+dl+'${environmentParamStr}';f.parentNode.insertBefore(j,f);
+  })(window,document,'script','dataLayer', '${id}');`
+
+const generateGTMIframe = ({ id, environmentParamStr }) =>
+  oneLine`<iframe src="https://www.googletagmanager.com/ns.html?id=${id}${environmentParamStr}" height="0" width="0" style="display: none; visibility: hidden"></iframe>`
+
+const generateDefaultDataLayer = (dataLayer, reporter) => {
+  let result = `window.dataLayer = window.dataLayer || [];`
+
+  if (typeof dataLayer === `function`) {
+    result += `window.dataLayer.push((${dataLayer})()));`
```

Comment:
Aren't there way too much brackets at the end? Or am I missing something here?

Reply:
good catch 1 to many 😄